html/modules/custom/reliefweb_api/src/Commands/ReliefWebApiCommands.php

```diff
@@ -300,4 +298,68 @@ public function replace($bundle, $newtag, $oldtag = NULL, array $options = [
     }
   }
 
+  /**
+   * Re-index queued content.
+   *
+   * @command reliefweb-api:reindexqueue
+   *
+   * @usage reliefweb-api:reindexqueue
+   *   Re-index the queue of terms to update.
+   *
+   * @validate-module-enabled reliefweb_api
+   *
+   * @aliases rw-api:reindexqueue
+   */
+  public function reIndexQueue() {
+
+    $reindex_queue = json_decode($this->state->get('reliefweb_api.reindex_queue'));
+
+    if (empty($reindex_queue)) {
+      return;
+    }
+
+    // Use default values from $this->index().
+    $options = [
+      'elasticsearch' => '',
+      'base-index-name' => '',
+      'website' => '',
+      'limit' => 0,
+      'offset' => 0,
+      'filter' => '',
+      'chunk-size' => 500,
+      'tag' => '',
+      'id' => 0,
+      'remove' => FALSE,
+      'replace' => FALSE,
+      'alias' => FALSE,
+      'alias-only' => FALSE,
+      'log' => 'echo',
+      'count-only' => FALSE,
+      'memory-limit' => '512M',
+      'replicas' => NULL,
+      'shards' => NULL,
+    ];
+    foreach (array_keys($reindex_queue) as $bundle) {
+      if (empty($reindex_queue[$bundle])) {
+        continue;
+      }
+      // Launch the re-indexing.
+      try {
+        $options['filter'] = $bundle . ':' . implode(',', $reindex_queue[$bundle]);
+        if ($bundle == 'country' || $bundle == 'source') {
+          $this->index('training', $options);
+          $this->index('job', $options);
+        }
+        $this->index('report', $options);
+      }
+      catch (\Exception $exception) {
+        $this->logger->error('(' . $exception->getCode() . ') ' . $exception->getMessage());
+      }
+      $reindex_queue[$bundle] = [];
+    }
+
+    $this->state->set('reliefweb_api.reindex_queue', json_encode($reindex_queue));
```

Comment:
```suggestion
    }
```

To go with the suggestion above that empties the queue as soon as it's retrieved.